sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaSuite.scala

```diff
@@ -114,7 +130,66 @@ abstract class ParquetSchemaTest extends ParquetTest with SharedSparkSession {
       sqlSchema,
       parquetSchema,
       binaryAsString,
-      int96AsTimestamp)
+      int96AsTimestamp,
+      expectedParquetColumn = expectedParquetColumn)
+  }
+
+  protected def compareParquetColumn(actual: ParquetColumn, expected: ParquetColumn): Unit = {
+    assert(actual.sparkType == expected.sparkType, "sparkType mismatch: " +
+        s"actual = ${actual.sparkType}, expected = ${expected.sparkType}")
+    assert(actual.descriptor === expected.descriptor, "column descriptor mismatch: " +
+        s"actual = ${actual.descriptor}, expected = ${expected.descriptor})")
+    // Parquet ColumnDescriptor equals only compare path so we'll need to compare other fields
```

Comment:
Shall we codify `equals` and `path`? And, maybe `compares` instead of `compare`?
```
- Parquet ColumnDescriptor equals only compare path so we'll need ...
- Since Parquet ColumnDescriptor `equals` only compares `path`, we need ...
```

Reply:
I think `path` equality is already compared above `actual.descriptor === expected.descriptor`? Fixed the comments.